lib/asm/arm64/Arm64Assembler.v3

```diff
@@ -0,0 +1,98 @@
+// Copyright 2024 Virgil Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+// Arm64 registers come in 32-bit and 64-bit variants
+type RegSize {
+	case Half;
+	case Full;
+}
+
+// Base class for all Arm64 registers
+// regnum is an ID 0-32
+class Arm64Reg(regnum: byte) {
+	def low3 = byte.view(regnum & 0b111);
+}
+
+// Representation of an Arm64 general purpose register (x0-x30),
+// or the zero register (xzr), or the stack pointer (sp).
+// An "int reg", is not a standard name in the Arm64 reference, but we use it
+// to distinguish from the SIMD and floating-point register file
+class Arm64IntReg extends Arm64Reg {
```

Comment:
Let's stick with `Gpr` because these registers can clearly hold pointers, and that fits better with Virgil's existing assemblers.

Reply:
Fixed